spec/requests/api/graphql/mutations/create_consignment_offer_spec.rb

```diff
@@ -0,0 +1,207 @@
+        JWT.encode(payload, Convection.config.jwt_secret)
+      end
+
+      it 'returns an error for that request' do
+        post '/api/graphql', params: { query: mutation }, headers: headers
+
+        expect(response.status).to eq 200
+        body = JSON.parse(response.body)
+
+        create_response = body['data']['createConsignmentOffer']
+        expect(create_response).to eq nil
+
+        error_message = body['errors'][0]['message']
+        expect(error_message).to eq "Can't access createConsignmentOffer"
+      end
+    end
+
+    context 'with a request missing a gravity partner id' do
+      let(:mutation_inputs) do
+        <<-INPUTS
+        {
+          submissionId: #{submission.id},
+          commissionPercentWhole: 10
+        }
+        INPUTS
+      end
+
+      it 'returns an error for that request' do
+        post '/api/graphql', params: { query: mutation }, headers: headers
+
+        expect(response.status).to eq 200
+        body = JSON.parse(response.body)
+
+        error_message = body['errors'][0]['message']
+        expect(error_message).to match 'is required'
+      end
+    end
+
+    context 'with a request missing a submission id' do
+      let(:mutation_inputs) do
+        <<-INPUTS
+        {
+          gravityPartnerId: "#{partner.gravity_partner_id}",
+          commissionPercentWhole: 10
+        }
+        INPUTS
+      end
+
+      it 'returns an error for that request' do
+        post '/api/graphql', params: { query: mutation }, headers: headers
+
+        expect(response.status).to eq 200
+        body = JSON.parse(response.body)
+
+        error_message = body['errors'][0]['message']
+        expect(error_message).to match 'is required'
+      end
+    end
+
+    context 'with a request missing a commission percent' do
+      let(:mutation_inputs) do
+        <<-INPUTS
+        {
+          gravityPartnerId: "#{partner.gravity_partner_id}",
+          submissionId: #{submission.id}
+        }
+        INPUTS
+      end
+
+      it 'returns an error for that request' do
+        post '/api/graphql', params: { query: mutation }, headers: headers
+
+        expect(response.status).to eq 200
+        body = JSON.parse(response.body)
+
+        error_message = body['errors'][0]['message']
+        expect(error_message).to match 'is required'
+      end
+    end
+  end
+
+  describe 'valid requests' do
+    context 'with just the minimum' do
+      it 'creates an offer' do
+        stub_gravity_root
+        stub_gravity_user
+        stub_gravity_user_detail(email: 'michael@bluth.com')
+
+        expect {
+          post '/api/graphql', params: { query: mutation }, headers: headers
+        }.to change(Offer, :count).by(1)
+
+        expect(response.status).to eq 200
+        body = JSON.parse(response.body)
+        create_response = body['data']['createConsignmentOffer']
+
+        offer_response = create_response['consignmentOffer']
+        expect(offer_response).to include({ 'id' => be })
```

Comment:
Just a question about ruby code that I don't understand - what does this line mean? Specifically `=> be`?

Reply:
Good question, I ended up replacing this with a comparison using the offer's id because that seemed more clear and would better lend itself to further expectations as we fill this out.

BUT! To your question, here's what's up - that `offer_response` is a hash. That means rspec gives us the `include` matcher. That matcher accepts an argument that will be checked only for what it includes. This means that offer response may have lots of keys, but it must at least has a key called id with a value. The last bit of magic is that `be` part - this is another magic bit from rspec and is basically saying that it's not nil. 😝